build_runner/lib/src/package_graph/apply_builders.dart

```diff
@@ -79,6 +91,9 @@ BuilderApplication apply(String builderKey,
         {bool isOptional,
         bool hideOutput,
         InputSet defaultGenerateFor,
+        BuilderOptions defaultOptions,
```

Comment:
After adding these arguments it might be better to take a `TargetBuilderConfigDefaults` rather than all of it's fields separately... but that would also make it a breaking change.